packages/core/src/rest/ZosmfRestClient.ts

```diff
@@ -51,53 +54,103 @@ export class ZosmfRestClient extends RestClient {
      * @memberof ZosmfRestClient
      */
     protected processError(original: IImperativeError): IImperativeError {
-        original.msg = "z/OSMF REST API Error:\n" + original.msg;
+        // TODO:V3_ERR_FORMAT - Remove block in V3
+        if (!NextVerFeatures.useV3ErrFormat()) {
+            original.msg = "z/OSMF REST API Error:\n" + original.msg;
+        }
 
-        let details = original.causeErrors;
+        let causeErrorsString = original.causeErrors; // TODO:V3_ERR_FORMAT - Remove in V3
+        let causeErrorsJson;
         try {
-            const json = JSON.parse(details);
-            // if we didn't get an error trying to parse json, check if there is a stack
-            // on the JSON error and delete it
-            if (json.stack != null) {
-                this.log.error("An error was encountered in z/OSMF with a stack." +
-                    " Here is the full error before deleting the stack:\n%s", JSON.stringify(json));
-                this.log.error("The stack has been deleted from the error before displaying the error to the user");
-                delete json.stack; // remove the stack field
-            }
+            // don't try to parse an empty string
+            if (original.causeErrors !== "") {
```

Comment:
Should we also handle the case of `original.causeErrors === undefined`, since "causeErrors" appears to be an optional property of IImperativeError?

Reply:
This logic existed in the original implementation. With all of the complication of the dual-path logic, I made an effort to NOT change any existing logic. I can see the concern. I will add guards for what could be existing errors.